src/metrics.rs

```diff
@@ -0,0 +1,117 @@
+//! raftlog のメトリクス。
+
+use prometrics::metrics::{Counter, Gauge, Histogram, HistogramBuilder, MetricBuilder};
+
+use node::NodeId;
+use {Error, Result};
+
+/// `raftlog` 全体に関するメトリクス。
+#[derive(Clone)]
+pub struct RaftlogMetrics {
+    pub(crate) node_state: NodeStateMetrics,
+}
+impl RaftlogMetrics {
+    pub(crate) fn new(builder: &MetricBuilder, node: &NodeId) -> Result<Self> {
+        let node_state = track!(NodeStateMetrics::new(builder, node))?;
+        Ok(Self { node_state })
+    }
+}
+
+/// ノード状態に関するメトリクス。
+#[derive(Clone)]
+pub struct NodeStateMetrics {
+    pub(crate) transit_to_candidate_total: Counter,
+    pub(crate) transit_to_follower_total: Counter,
+    pub(crate) transit_to_leader_total: Counter,
+    pub(crate) event_queue_len: Gauge,
+    pub(crate) poll_timeout_total: Counter,
+    pub(crate) candidate_to_leader_duration_seconds: Histogram,
+    pub(crate) candidate_to_follower_duration_seconds: Histogram,
+    pub(crate) loader_to_candidate_duration_seconds: Histogram,
+}
+impl NodeStateMetrics {
+    pub(crate) fn new(builder: &MetricBuilder, node: &NodeId) -> Result<Self> {
+        let node = node.as_str();
+        let mut builder: MetricBuilder = builder.clone();
+        builder.subsystem("node_state");
+        let transit_to_candidate_total = track!(builder
+            .counter("transit_to_candidate_total")
+            .help("Number of transitions to candidate role")
+            .label("node", &node)
```

Comment:
I think that it should be avoided to always append `NodeId` as an label for the same reason as https://github.com/frugalos/frugalos/pull/138/files#r272589593.

Reply:
Thank you for reviewing. I will remove `NodeId` from labels. The reason is described at https://github.com/frugalos/frugalos/pull/138/files#r272778388.

Reply:
fixed